### PR TITLE
Make heart progress bar on fundraising page

### PR DESCRIPTION
**Pull request by treyhunner**
This is an attempt at a progress bar heart for #293.

How it works:

1. Inline SVG element is used with a `switch` that will use the PNG image as a fallback on older browsers ([per suggestion here](http://davidensinger.com/2013/04/inline-svg-with-png-fallback/))
2. Array of pixel coordinates/colors is stored in JavaScript and used to render rectangles for the "pixels"
3. Random "pixels" are removed from the heart based on the percentage of the fundraising goal attained
4. An outline is displayed around the heart to show what the shape will look like once the fundraising goal is reached

A few ideas for functionality changes:

1. Show pixels appearing as the page loads or moving while on the page
2. Show missing pixels in another color or show them with a very low opacity level so they're near-white
3. Remove outline path or make it look better (the black is a little ugly I think)

The JS code should also probably be wrapped in a Modernizr check for SVG support should.  I have not tested the fallback image in IE8 but I tested it on an old Android phone and the PNG is still shown.

Thoughts on functional and/or implementation improvements?

### CODE REVIEW

**Comment by treyhunner** [comment]
Here's two screenshots of the random missing pixels as they currently appear

selection_259
selection_260

---

Here's a picture of the heart on an old Android device that doesn't support SVG
2015-01-15 20 56 34-cropped

**Comment by treyhunner** [comment]
My last few commits changed the missing pixels to faded pixels on a light gray background with a soft border.  A faded pixel moves its place every 5 seconds and there's a slight animation on pixel fading.  Hovering over faded pixels also lights them up some.

I'm going to stop fiddling with this for now before I get too carried away (if I haven't already).

All input welcome.

**Comment by olasitarska** [comment]
Wow @treyhunner, no need to apologize at all. This looks **amazing**! :clap: 

@jezdez any objections on merging it?:)

**Comment by asendecka** [comment]
Yes, it looks awesome! :)

**Comment by jezdez** [comment]
No objections :blue_heart:!

File: djangoproject/templates/fundraising/index.html [hunk 1]
```diff
@@ -13,8 +13,17 @@ <h1 class="visuallyhidden">Support Django</h1>
...
-  <div class="heart">
-    <img src="{% static 'img/fundraising-heart.png' %}" />
+  <div class="heart" data-percent="51">
+    <svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 503 431">
+      <switch>
+        <g transform="translate(4, 2)" id="pixels">
+          <path d="M 71 0 L 213 0 L 213 71 L 284 71 L 284 0 L 426 0 L 426 71 L 497 71 L 497 213 L 426 213 L 426 284 L 355 284 L 355 355 L 284 355 L 284 426 L 213 426 L 213 355 L 142 355 L 142 284 L 71 284 L 71 213 L 0 213 L 0 71 L 71 71 L 71 0" fill="#f8f8f8" stroke="#c0c0c0" stroke-width="1" />
+        </g>
+        <foreignObject>
+          <img src="{% static 'img/fundraising-heart.png' %}" />
```
Comment (MarkusH) [inline djangoproject/templates/fundraising/index.html]: Does this file exist anywhere?

Comment (olasitarska) [inline djangoproject/templates/fundraising/index.html]: I think so! I added it to fundraising branch already :)

**Comment by olasitarska** [comment]
:heart: :blue_heart: :purple_heart: :yellow_heart: :green_heart:

File: djangoproject/templates/fundraising/index.html (hunk 1 above)
Comment (MarkusH) [inline djangoproject/templates/fundraising/index.html]: Oh. Missed that.

Comment (olasitarska) [inline djangoproject/templates/fundraising/index.html]: :+1:

**Pull request merged**